test/cql-pytest/test_tablets.py

```diff
@@ -121,3 +121,49 @@ def test_tablets_are_dropped_when_dropping_table(cql, skip_without_tablets):
             assert res.one().tablet_count > 0, f"table {keyspace_name}.{table_name}: zero tablets allocated"
         res = cql.execute(f"SELECT * FROM system.tablets WHERE keyspace_name='{keyspace_name}' AND table_name='{table_name}' ALLOW FILTERING")
         assert not res, f"table {keyspace_name}.{table_name} was found in system.tablets after the table was dropped: tablet_count={res.one().tablet_count}"
+
+
+def test_tablets_are_dropped_when_dropping_view(cql, skip_without_tablets):
+    ksdef = "WITH REPLICATION = {'class' : 'NetworkTopologyStrategy', 'replication_factor' : '1'} AND TABLETS = {'initial': 1}"
+    with new_test_keyspace(cql, ksdef) as keyspace:
+        keyspace_name = keyspace
+        table_name = ""
+        view_name = ""
+        with new_test_table(cql, keyspace, "pk int PRIMARY KEY, c int") as table:
+            table_name = table.split('.')[1]
+            with new_materialized_view(cql, table, '*', 'c, pk', 'c is not null and pk is not null') as mv:
+                view_name = mv.split('.')[1]
+                for name in [table_name, view_name]:
+                    desc = "table" if name == table_name else "view"
+                    res = cql.execute(f"SELECT * FROM system.tablets WHERE keyspace_name='{keyspace_name}' AND table_name='{name}' ALLOW FILTERING")
+                    assert res is not None, f"{desc} {keyspace_name}.{name} not found in system.tablets after view is created"
+                    assert res.one().tablet_count > 0, f"{desc} {keyspace_name}.{name}: zero tablets allocated"
+            res = cql.execute(f"SELECT * FROM system.tablets WHERE keyspace_name='{keyspace_name}' AND table_name='{view_name}' ALLOW FILTERING")
+            assert not res, f"view {keyspace_name}.{view_name} was found in system.tablets after the view was dropped: tablet_count={res.one().tablet_count}"
+        res = cql.execute(f"SELECT * FROM system.tablets WHERE keyspace_name='{keyspace_name}' AND table_name='{table_name}' ALLOW FILTERING")
+        assert not res, f"table {keyspace_name}.{table_name} was found in system.tablets after the view was dropped: tablet_count={res.one().tablet_count}"
+
+
+def test_tablets_are_not_dropped_when_dropping_table_with_view_fails(cql, skip_without_tablets):
+    ksdef = "WITH REPLICATION = {'class' : 'NetworkTopologyStrategy', 'replication_factor' : '1'} AND TABLETS = {'initial': 1}"
+    with new_test_keyspace(cql, ksdef) as keyspace:
+        keyspace_name = keyspace
+        table_name = ""
+        view_name = ""
+        with pytest.raises(InvalidRequest):
```

Comment:
I had to read this test several times to understand that you expect the new_test_table() function itself to fail, because you created a view in this table and don't remove it, so the table deletion will fail. I think this is not a good use of new_test_table() and you could have just used CREATE TABLE directly (anyway, you lost the protection of deleting the table on exception - the table will be deleted when you delete the keyspace (this test really assumes a new keyspace and not a fixture).

But more importantly, I don't understand why this test needs to exist in the first place -  the DROP TABLE got refused, so nothing should have happened. Testing the tablets didn't get deleted is strange given that nothing at all should have happened (neither the table nor the view changed in any way, didn't get deleted, their tablets didn't, and so on).

Reply:
> the DROP TABLE got refused, so nothing should have happened.

should => worth verification

Reply:
fixed in next version by using internal test functions accepting flags for edge-cases
I hope that would be clearer